src/ageml/ui.py

```diff
@@ -184,6 +290,289 @@ def load_csv(self, file_type):
+        elif [df[col].dtype == bool for col in df.columns].count(False) != 0:
+            raise TypeError("Clinical columns must be boolean type. Check that all values are encoded as 'True' or 'False'.")
+        
+        # Check that all columns have at least two subjects and show which column
+        for col in df.columns:
+            if df[col].sum() == 0:
+                raise ValueError("Clinical column %s has no subjects." % col)
+
+        # Find rows with all False
+        if not df.any(axis=1).all():
+            raise ValueError("Clinical file contains rows with all False values. Please check the file.")
+
+        # Set clinical flag
+        self.flags['clinical'] = True
+
+        return df
+
+    def load_factors(self, required=False):
+        """Load factors from csv file.
+
+        Parameters
+        ----------
+        required: boolean to check if file is required
+
+        Returns
+        -------
+        df: dataframe with factors"""
+        
+        # Load data
+        df = self.load_csv('factors')
+
+        # Check required
+        if required and df is None:
+            raise ValueError("Factors file must be provided.")
+        elif df is None:
+            return df
+        
+        return df
+
+    def load_ages(self, required=False):
+        """Load ages from csv file.
+
+        Parameters
+        ----------
+        required: boolean to check if file is required
+
+        Returns
+        -------
+        df: dataframe with ages"""
+
+        # Load data
+        df = self.load_csv('ages')
+
+        # Check required
+        if required and df is None:
+            raise ValueError("Ages file must be provided.")
+        elif df is None:
+            return df
+
+        # Required columns
+        self.flags['ages'] = True
+        req_cols = ["age", "predicted_age", "corrected_age", "delta"]
+        cols = [col.lower() for col in df.columns.to_list()]
+
+        # Check that columns are present
+        for col in req_cols:
+            if not any(c.startswith(col) for c in cols):
+                raise KeyError("Ages file missing the following column %s, or derived names." % col)
+
+        # Check that columns present are of the correct type
+        for col in cols:
+            if not any(col.startswith(c) for c in req_cols):
+                raise KeyError("Ages file contains unknwon column %s" % col)
+
+        return df
+
+    def system_parser(self, file):
+        """Parse systems file.
+
+        Parameters
+        ----------
+        file: file to parse
+
+        Returns
+        -------
+        dict_systems: dictionary with systems and their features"""
+
+        # Initialise dictionary
+        systems = {}
+
+        # Load feature names
+        features = {f.lower() for f in self.df_features.columns.to_list()}
+
+        # Parse file
+        for line in open(file, 'r'):
+            line = line.split("\n")[0]  # Remove newline character
```

Comment:
This will likely cause a bug in Windows generated files, because the Windows newline character is `\n\r`